### PR TITLE
Add strippable block data map

### PR DESCRIPTION
**Pull request by Spinoscythe**
Adds the strippable block data map

### CODE REVIEW

**Comment by KnightMiner** [comment]
Any reason not to also do the shovel version of this? I know hoes are a little more complex due to rooted dirt but unless things changed recently shovels should be just another map.

**Comment by Spinoscythe** [comment]
I thought of making one for shovels and hoes, and thought I should make separate PRs for them. it would be easier to track back

**Comment by Technici4n** [comment]
This needs careful consideration when only one side has data maps.

**Comment by Matyrobbrt** [comment]
You need to be careful about data maps like these. First of all, this won't work for blocks that don't use that exact property for rotation. Additionally, the tool plays the sound on the client for the current player, so a neo client connected to a vanilla server will not have sounds played for any strip, and a vanilla client to a neo server will not have sounds for custom strippables.

**Comment by Nekretaur** [comment]
@Matyrobbrt What solutions would you recommend for this?

File: patches/net/minecraft/world/item/AxeItem.java.patch [hunk 1]
```diff
@@ -42,20 +43,20 @@
...
                      p_308922_.levelEvent(p_309192_, 3004, p_308899_, 0);
-@@ -99,8 +_,19 @@
+@@ -99,8 +_,18 @@
          }
      }
  
 +    public static BlockState getAxeStrippingState(BlockState originalState) {
 +        Block block = originalState.getBlock().builtInRegistryHolder().getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.STRIPPABLES).strippedBlock();
-+        return block.defaultBlockState().setValue(RotatedPillarBlock.AXIS, originalState.getValue(RotatedPillarBlock.AXIS));
++        return block.defaultBlockState();
```
Comment (KnightMiner) [inline patches/net/minecraft/world/item/AxeItem.java.patch:52]: Isn't there a blockstate method that copies all matching properties from the source to the definition? Might be worth using that here.

Comment (Spinoscythe) [reply patches/net/minecraft/world/item/AxeItem.java.patch:52]: IDK there is one AFAIK

**Comment by Spinoscythe** [comment]
> You need to be careful about data maps like these. First of all, this won't work for blocks that don't use that exact property for rotation. Additionally, the tool plays the sound on the client for the current player, so a neo client connected to a vanilla server will not have sounds played for any strip, and a vanilla client to a neo server will not have sounds for custom strippables. 

do you have any suggestions @Matyrobbrt ?

**Pull request closed without merging**